themes/docs-new/layouts/shortcodes/node_attribute.md

```diff
@@ -1,35 +1,22 @@
-An attribute is a specific detail about a node. Attributes are used by
-Chef Infra Client to understand:
+An attribute is a specific detail about a node. Attributes are used by Chef Infra Client to understand:
 
--   The current state of the node
--   What the state of the node was at the end of the previous Chef Infra
-    Client run
--   What the state of the node should be at the end of the current Chef
-    Infra Client run
+- The current state of the node
+- What the state of the node was at the end of the previous Chef Infra Client run
+- What the state of the node should be at the end of the current Chef Infra Client run
 
 Attributes are defined by:
 
--   The state of the node itself
--   Attributes passed via JSON on the CLI
--   Cookbooks (in attribute files and/or recipes)
--   Roles
--   Environments
--   Policyfiles
+- The node as saved on the Chef Infra Server
```

Comment:
Maybe I don't understand, but this seems circular to me. It is like saying a node sets its own attributes. My understanding is that node that hasn't had a Chef Infra Client run has no attributes, and it is only after a run that the node object with default attributes exists for Chef Infra Server to store. The Chef Infra Client run uses the node object as an index, against which it compares existing node content against content that it needs to download. Beyond comparing the list of files that already exist on the server, the Chef Infra Client removes and then rewrites the node object. Or am I missing something?